test/parallel/test-http2-client-rststream-before-connect.js

```diff
@@ -12,6 +12,12 @@ server.on('stream', (stream) => {
   stream.on('close', common.mustCall());
   stream.respond();
   stream.end('ok');
+
+  stream.on('error', common.expectsError({
+    code: 'ERR_HTTP2_STREAM_ERROR',
+    name: 'Error',
+    message: 'Stream closed with error code NGHTTP2_PROTOCOL_ERROR'
+  }));
```

Comment:
I'm not sure why/how this occurs with this PR and whether this is a bug or missing test logic? @bnoordhuis @addaleax

Reply:
Basically it occurs when `'finish'` is delayed by an extra `nextTick`. https://github.com/nodejs/node/pull/32780/files#diff-1915a7b992a3012b177dd855fa3477e0R645